app/src/main/java/com/github/se/polyfit/viewmodel/post/CreatePostViewModel.kt

```diff
@@ -60,8 +63,18 @@ constructor(
     _post.createdAt = createdAt
   }
 
-  fun setPostLocation(location: Location) {
-    _post.location = location
+  fun setPostLocation(): Job {
+    return viewModelScope.launch(Dispatchers.IO) {
+      val location = postLocationModel.getCurrentLocation()
+      _post.location = location
+    }
+  }
+
+  fun setInOrder(first: Job = setPostLocation(), second: () -> Unit = ::setPost) {
```

Comment:
Maybe it would be better to remove the arguments and directly call the functions in the body of this function, and give it a more suitable name.

Reply:
same It's just for testing the function,otherwise I don't manage to catch the execution of the function in the tests

Reply:
Do you mean that if you have a coroutine inside a ViewModel scope, you weren't able to catch the error when mocking it?